staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -329,6 +329,10 @@ type Cacher struct {
 	expiredBookmarkWatchers []*cacheWatcher
 }
 
+func (c *Cacher) RequestBookmark(ctx context.Context) error {
```

Comment:
I failed to find a comparison testing between storage and watchcache. Could you point me to exact tests? Also what this test would be validating? That both cacher and storage call etcdclient? Mocking etcd client seems like a overkill.

 I also don't think this test adds anything, would prefer to separate etcd3.storage struct from storage.Interface. Others consumers of don't need and won't use storage.Interface.

Reply:
> I failed to find a comparison testing between storage and watchcache. Could you point me to exact tests?

We don't have comparison per-se. We have a set of tests that we run against both watchcache and etcd3.
These are all tests implemented here:
https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/apiserver/pkg/storage/testing


> Also what this test would be validating? That both cacher and storage call etcdclient? Mocking etcd client seems like a overkill.

No - we want to validate the semantic. I don't want to mock anything.
I would be fine with observing the bookmark itself. Or maybe even just ensuring that LIST returns the right thing. But I want a test.


> I also don't think this test adds anything, would prefer to separate etcd3.storage struct from storage.Interface. Others consumers of don't need and won't use storage.Interface.

I don't understand what you mean. Can you clarify?

Reply:
> > I also don't think this test adds anything, would prefer to separate etcd3.storage struct from storage.Interface. Others consumers of don't need and won't use storage.Interface.
> 
> I don't understand what you mean. Can you clarify?

I would prefer to avoid to expose `RequestBookmark` from the watchcache. Didn't do that for now as it would be a bigger change, but semantically it makes no sense that `storage.Interface` exposes a function that is just used for internally.

Reply:
OK - but then let's make it clear in the interface that it's a temporary thing and we would like to get rid of it.

Reply:
Either mark it as DEPRECATED or sth.

Reply:
Done